executor/poolmgr/gp.go

```diff
@@ -297,15 +297,28 @@ func (gp *GenericPool) getFetcherUrl(podIP string) string {
 	return fmt.Sprintf("http://%v:8000/", podIP)
 }
 
+func IsIPv6(podIP string) bool {
+	ip := net.ParseIP(podIP)
+	return ip != nil && strings.Contains(podIP, ":")
+}
+
 func (gp *GenericPool) getSpecializeUrl(podIP string, version int) string {
 	u := os.Getenv("TEST_SPECIALIZE_URL")
 	if len(u) != 0 {
 		return u
 	}
-	if version == 1 {
-		return fmt.Sprintf("http://%v:8888/specialize", podIP)
+	t := IsIPv6(podIP)
```

Comment:
Thanks, @valentin2105 !

May I suggest reorganizing this if-else to set the host part of the URL separately from the version check? That way you won't need the nested if-s.